lib/web/mage/translate.js

```diff
@@ -38,7 +38,7 @@ define([
                      * @return {String}
                      */
                     translate: function (text) {
-                        return _data[text] ? _data[text] : text;
+                        return _data.hasOwnProperty(text) ? _data[text] : text;
```

Comment:
This could break if the `_data` dict is ever created via `Object.create(null)` (used sometimes for dicts in JS).

I'd recommend being more explicit, and doing `typeof _data[text] === 'string'` instead of `hasOwnProperty`. Although I don't expect folks to be messing with the prototypes of these objects, we do have people that generate their own dicts outside of static content deploy. It makes more sense for us to check the property in a way that will also respect the prototype chain.

Reply:
What if we just check that value exists e.g.
`typeof _data[text] !== 'undefined'` ?

Reply:
That works too :) I'd be happy with either